lib/rspec/support/differ.rb

```diff
@@ -7,8 +7,14 @@ module RSpec
   module Support
     # rubocop:disable ClassLength
     class Differ
+      if String.method_defined?(:encoding)
+        EMPTY_DIFF = EncodedString.new("", Encoding.default_external)
+      else
+        EMPTY_DIFF = EncodedString.new("")
+      end
+
       def diff(actual, expected)
-        diff = ""
+        diff = EMPTY_DIFF.dup
```

Comment:
Why was `""` insufficient?

Also, is the `dup` just to guard against mutations (to ensure `EMPTY_DIFF` isn't mutated)?  If there aren't any mutations maybe you could freeze the constant instead and use it w/o duping for fewer allocations.